pkg/skaffold/runner/deployer.go

```diff
@@ -95,6 +111,10 @@ func GetDeployer(runCtx *runcontext.RunContext, provider deploy.ComponentProvide
 		}
 	}
 
+	if localDeploy && remoteDeploy {
+		return nil, errors.New("docker deployment not supported alongside cluster deployments")
+	}
```

Comment:
Is there a technical reason for this?

Reply:
not technically. but I think we should put this guardrail in place while we let this feature develop. once we've identified some real use cases for spinning up local containers alongside cluster deployments, I'd be willing to explore how skaffold enables that, but I would expect unexpected behavior if we just blindly allow it.